### PR TITLE
change the fork base from master to restruct

### PR DESCRIPTION
**Pull request by jalynnliu**

### CODE REVIEW

**Comment by jalynnliu** [comment]
加 dropout
改 evalute 接口和存储模型的方式
添加 learning_rate 种类
结果很差返回-1
